src/config/ConfigManager.cpp

```diff
@@ -55,19 +67,62 @@ static std::expected<std::string, std::string> resolvePath(const std::string_vie
     return CAN;
 }
 
-static std::expected<std::string, std::string> getFullPath(const std::string_view& sv) {
-    if (sv.empty())
+static std::expected<std::string, std::string> getPath(const std::string_view& path) {
+    if (path.empty()) {
         return std::unexpected("empty path");
+    }
 
-    if (sv[0] == '~') {
+    if (path[0] == '~') {
         static auto HOME = getenv("HOME");
         if (!HOME || HOME[0] == '\0')
             return std::unexpected("home path but no $HOME");
 
-        return resolvePath(std::string{HOME} + "/"s + std::string{sv.substr(1)});
+        return resolvePath(std::string{HOME} + "/"s + std::string{path.substr(1)});
     }
 
-    return resolvePath(sv);
+    return resolvePath(path);
+}
+
+static std::expected<std::vector<std::string>, std::string> getFullPath(const std::string& sv) {
+    if (sv.empty())
+        return std::unexpected("empty path");
+
+    static constexpr const size_t maxImagesCount{100};
+
+    const auto                    paths = Hyprutils::String::CConstVarList(sv);
+    std::vector<std::string>      result;
+    result.reserve(paths.size());
+
+    for (const auto& path : paths) {
+        if (result.size() >= maxImagesCount) {
+            g_logger->log(LOG_WARN, "Maximum number of images ({}) reached", maxImagesCount);
+            break;
+        }
+
+        const auto resolved = getPath(path);
+        if (!resolved) {
+            return std::unexpected(resolved.error());
+        }
+
+        const auto resolvedPath = resolved.value();
+
+        if (std::filesystem::is_directory(resolvedPath)) {
+            for (const auto& entry : std::filesystem::directory_iterator(resolvedPath)) {
```

Comment:
don't recurse

Reply:
that was what I meant? If path is a file, one image. If it's a dir, all images in that dir (non recursively)

Reply:
yes that good